build-env docker support  mac arm

## Feature Request

**Is your feature request related to a problem? Please describe:**
<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->
some dev env is mac  with arm chip, so docker can support arm64.

ref https://docs.docker.com/build/ci/github-actions/multi-platform/,  arm64 docker can be supported.

**Describe the feature you'd like:**
<!-- A clear and concise description of what you want to happen. -->

**Describe alternatives you've considered:**
<!-- A clear and concise description of any alternative solutions or features you've considered. -->

**Teachability, Documentation, Adoption, Migration Strategy:**
<!-- If you can, explain some scenarios how users might use this, situations it would be helpful in. Any API designs, mockups, or diagrams are also helpful. -->

I build a docker for arm64, but some compile error happen

```
docker run -v `pwd`:/root/pegasus \
           xshalk/pegasus:build-env-ubuntu2204-20240722  \
           /bin/bash -c "cd /root/pegasus; ./run.sh build -c --clear_thirdparty -j $(nproc)"
```

log 
```
gcc.compile.c++ bin.v2/libs/system/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/error_code.o
gcc.archive bin.v2/libs/system/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/libboost_system.a
common.copy lib/libboost_system.a
gcc.compile.c++ bin.v2/libs/filesystem/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/codecvt_error_category.o
gcc.compile.c++ bin.v2/libs/filesystem/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/operations.o
gcc.compile.c++ bin.v2/libs/filesystem/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/path.o
gcc.compile.c++ bin.v2/libs/filesystem/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/path_traits.o
gcc.compile.c++ bin.v2/libs/filesystem/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/portability.o
gcc.compile.c++ bin.v2/libs/filesystem/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/unique_path.o
gcc.compile.c++ bin.v2/libs/filesystem/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/utf8_codecvt_facet.o
gcc.compile.c++ bin.v2/libs/filesystem/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/windows_file_codecvt.o
gcc.archive bin.v2/libs/filesystem/build/gcc-11.4.0/release/cxxstd-11-iso/link-static/threading-multi/visibility-hidden/libboost_filesystem.a
common.copy lib/libboost_filesystem.a
...updated 14775 targets...
[ 65%] Performing install step for 'boost'
[ 66%] Completed 'boost'
[ 66%] Built target boost
make: *** [Makefile:91: all] Error 2
```

--- a/.github/workflows/build-push-env-docker.yml
+++ b/.github/workflows/build-push-env-docker.yml
@@ -52,19 +52,20 @@ jobs:
       - name: Checkout
         # The glibc version on ubuntu1804 and centos7 is lower than the actions/checkout@v4 required, so
         # we need to force to use actions/checkout@v3.
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v3
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+        uses: docker/setup-buildx-action@v3
       - name: Login to DockerHub
-        uses: docker/login-action@v1
+        uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKERHUB_USER }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
       - name: Build and push
-        uses: docker/build-push-action@v2.10.0
+        uses: docker/build-push-action@v6
         with:
+          platforms: linux/amd64,linux/arm64
           context: .
           file: ./docker/pegasus-build-env/${{ matrix.dockertag }}/Dockerfile
           push: true

